azure-mgmt-resources/src/main/java/com/azure/management/BearerTokenAuthenticationPolicy.java

```diff
@@ -76,21 +35,11 @@ private String getDefaultScopeFromRequest(HttpRequest request) {
             return Mono.error(new RuntimeException("token credentials require a URL using the HTTPS protocol scheme"));
         }
 
-        String[] scopes;
-        if (this.scopes == null || this.scopes.length == 0) {
-            scopes = new String[] {getDefaultScopeFromRequest(context.getHttpRequest())};
-        } else {
-            scopes = this.scopes;
-        }
-        assert scopes.length > 0;
-
         Mono<AccessToken> tokenResult;
-        AccessToken token = tokenCache.get(scopes[0]);
-        if (token == null || token.isExpired()) {
-            tokenResult = this.credential.getToken(new TokenRequestContext().addScopes(scopes))
-                            .doOnNext(accessToken -> this.tokenCache.put(scopes[0], accessToken));
+        if (this.scopes == null || this.scopes.length == 0) {
+            tokenResult = this.credential.getToken(context.getHttpRequest());
         } else {
-            tokenResult = Mono.just(token);
+            tokenResult = this.credential.getToken(new TokenRequestContext().addScopes(scopes));
```

Comment:
After multiple PR updates, It is now a bit messed up on the scope vs. request (host).

There is 3 thing happening here:
(1) If scope is not provided, deduct the scope from host+environment
(2) Get the token based on scope
(3) Cache the token

If we stick with scope, my suggestion:
(2) is always done in ApplicationTokenCredential. No problem.
(1) is moved a lot in various PRs. My suggestion is to put it to a Util class. Input host+environment, output scope. Deterministic, and easy to do unit test. No need to couple this to either TokenCredential or Policy.
After it is decoupled, Policy could then call TokenCredential.getToken with TokenRequestContext (use the Util above to get the scope if not provided).
(3) seems better to be in subclass of TokenCredential, since AuxiliaryAuthenticationPolicy would also use that cache.

@yaohaizh Let us know your opinion.

Reply:
From my perspective, I first use the TokenCredential for authorization, so every function should write outside the credential. But during the auxiliary authentication, I found that all projects are based on AzureTokenCredential, I think this common function could be written inside the AzureTokenCredential. 

Do we have scenario to use the TokenCredential rather than AzureTokenCredential?

Reply:
My consideration is, here you call Token on 2 cases, one is scope, another is request. So Token class got depends on two things, scope object or request object, based on things outside.

If we stick to scope, then scope object is a must. However request object is not a must have, since any static function could calculate scope from host+environment (a function which not related to Token class at all, since we just moved it multiple times now), then just provide scope to Token, which in turn make Token class only depends on scope object.

Anyway, it is up to you.

Reply:
Yeah, the calculation could use a Utils to calculate (I would fix it). And the cache, I think it couldn't be based on the TokenCredential, which actually makes all credential check harder. I'll see if it is better to use a cache token in the AzureTokenCredential.

Reply:
Cache should be fine at current state.

Do we now still need AzureTokenCredential in RestClient? Or TokenCredential is enough?
Ask because found this code in track1, the FunctionCredentials extends TokenCredenttial
```
            functionService = manager().restClient().newBuilder()
                    .withBaseUrl(defaultHostName.toString())
                    .withCredentials(new FunctionCredentials(this))
                    .withLogLevel(LogLevel.BODY_AND_HEADERS)
                    .build()
                    .retrofit().create(FunctionService.class);
```

Not a big issue, I can handle it later when doing for appservice.

Reply:
I'll check it, since the tokenCredentials in track 1 is different from track 2

Reply:
I have revert the AzureTokenCredential commit, change it back to TokenCredential in restClient, and split the tokencache with lock